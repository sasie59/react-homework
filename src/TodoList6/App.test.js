import React from 'react';
import { getByTestId, render,screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import HW,{ALL,FINISH,UNFINISH} from './';

test('should todoList', () => {
  const { container, getByTestId } = render(<HW />);
  userEvent.type(container.querySelector('input' ),'abc{enter}');
  userEvent.type(container.querySelector('input' ),'123{enter}');
  expect(getByTestId('todo-0').textContent).toBe('abctoggleremove');
  expect(getByTestId('todo-1').textContent).toBe('123toggleremove');
  
  userEvent.click(getByTestId('toggle-0'));
  expect(container).toMatchSnapshot();
  
  userEvent.click(getByTestId('remove-0'));
  expect(container).toMatchSnapshot();
});

test('should selectOptions', () => {
  const { container } = render(<HW />);
  userEvent.selectOptions(screen.getByTestId('select'),['ALL']);
  expect(container).toMatchSnapshot();
  userEvent.selectOptions(screen.getByTestId('select'),['FINISH']);
  expect(container).toMatchSnapshot();
  userEvent.selectOptions(screen.getByTestId('select'),['UNFINISH']);
  expect(container).toMatchSnapshot();
});

