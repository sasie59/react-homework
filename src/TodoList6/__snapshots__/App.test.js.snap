// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should selectOptions 1`] = `
<div>
  <div>
    <select
      data-testid="select"
    >
      <option
        data-testid="all"
        value="ALL"
      >
        ALL
      </option>
      <option
        data-testid="finish"
        value="FINISH"
      >
        FINISH
      </option>
      <option
        data-testid="unFinish"
        value="UNFINISH"
      >
        UNFINISH
      </option>
    </select>
    <form>
      <input
        type="text"
        value=""
      />
      <button>
        add
      </button>
    </form>
    <div
      class=""
      data-testid="todo-0"
    >
      123
      <button
        data-testid="toggle-0"
      >
        toggle
      </button>
      <button
        data-testid="remove-0"
      >
        remove
      </button>
    </div>
  </div>
</div>
`;

exports[`should selectOptions 2`] = `
<div>
  <div>
    <select
      data-testid="select"
    >
      <option
        data-testid="all"
        value="ALL"
      >
        ALL
      </option>
      <option
        data-testid="finish"
        value="FINISH"
      >
        FINISH
      </option>
      <option
        data-testid="unFinish"
        value="UNFINISH"
      >
        UNFINISH
      </option>
    </select>
    <form>
      <input
        type="text"
        value=""
      />
      <button>
        add
      </button>
    </form>
  </div>
</div>
`;

exports[`should selectOptions 3`] = `
<div>
  <div>
    <select
      data-testid="select"
    >
      <option
        data-testid="all"
        value="ALL"
      >
        ALL
      </option>
      <option
        data-testid="finish"
        value="FINISH"
      >
        FINISH
      </option>
      <option
        data-testid="unFinish"
        value="UNFINISH"
      >
        UNFINISH
      </option>
    </select>
    <form>
      <input
        type="text"
        value=""
      />
      <button>
        add
      </button>
    </form>
    <div
      class=""
      data-testid="todo-0"
    >
      123
      <button
        data-testid="toggle-0"
      >
        toggle
      </button>
      <button
        data-testid="remove-0"
      >
        remove
      </button>
    </div>
  </div>
</div>
`;

exports[`should todoList 1`] = `
<div>
  <div>
    <select
      data-testid="select"
    >
      <option
        data-testid="all"
        value="ALL"
      >
        ALL
      </option>
      <option
        data-testid="finish"
        value="FINISH"
      >
        FINISH
      </option>
      <option
        data-testid="unFinish"
        value="UNFINISH"
      >
        UNFINISH
      </option>
    </select>
    <form>
      <input
        type="text"
        value=""
      />
      <button>
        add
      </button>
    </form>
    <div
      class="line"
      data-testid="todo-0"
    >
      abc
      <button
        data-testid="toggle-0"
      >
        toggle
      </button>
      <button
        data-testid="remove-0"
      >
        remove
      </button>
    </div>
    <div
      class=""
      data-testid="todo-1"
    >
      123
      <button
        data-testid="toggle-1"
      >
        toggle
      </button>
      <button
        data-testid="remove-1"
      >
        remove
      </button>
    </div>
  </div>
</div>
`;

exports[`should todoList 2`] = `
<div>
  <div>
    <select
      data-testid="select"
    >
      <option
        data-testid="all"
        value="ALL"
      >
        ALL
      </option>
      <option
        data-testid="finish"
        value="FINISH"
      >
        FINISH
      </option>
      <option
        data-testid="unFinish"
        value="UNFINISH"
      >
        UNFINISH
      </option>
    </select>
    <form>
      <input
        type="text"
        value=""
      />
      <button>
        add
      </button>
    </form>
    <div
      class=""
      data-testid="todo-0"
    >
      123
      <button
        data-testid="toggle-0"
      >
        toggle
      </button>
      <button
        data-testid="remove-0"
      >
        remove
      </button>
    </div>
  </div>
</div>
`;
